azkaban-web-server/src/main/java/azkaban/imagemgmt/services/ImageTypeServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package azkaban.imagemgmt.services;
+
+import azkaban.imagemgmt.daos.ImageTypeDao;
+import azkaban.imagemgmt.exeception.ImageMgmtException;
+import azkaban.imagemgmt.exeception.ImageMgmtValidationException;
+import azkaban.imagemgmt.models.ImageType;
+import azkaban.imagemgmt.models.RequestContext;
+import azkaban.imagemgmt.utils.ValidatorUtils;
+import java.io.IOException;
+import javax.inject.Inject;
+import org.codehaus.jackson.map.ObjectMapper;
+
+public class ImageTypeServiceImpl implements ImageTypeService {
+
+  private final ImageTypeDao imageTypeDao;
+  private final ObjectMapper objectMapper;
+
+  @Inject
+  public ImageTypeServiceImpl(final ImageTypeDao imageTypeDao, final ObjectMapper objectMapper) {
+    this.imageTypeDao = imageTypeDao;
+    this.objectMapper = objectMapper;
+  }
+
+  @Override
+  public int createImageType(RequestContext requestContext) throws IOException,
+      ImageMgmtException {
+    ImageType imageType = objectMapper.readValue(requestContext.getJsonPayload(), ImageType.class);
+    imageType.setCreatedBy(requestContext.getUser());
+    if(!ValidatorUtils.validateObject(imageType)) {
```

Comment:
Will this validate correct value for enums like deployable or role?

Reply:
json string to object conversion using ObjectMapper will fail If the enum is incorrect.